### ISSUE TITLE
`github/async-preventdefault` should not flag calls before the first `await`

### ISSUE THREAD

**Issue by silverwind** [github/eslint-plugin-github#599]
`eventTarget.preventDefault()` works fine before the first `await` in an async function, so this code works as expected, but if currently flagged:

```js
document.querySelector('a').addEventListener('click', e => {
  e.preventDefault();
  await fetch('/');
});
```

It's only after the first `await`, where `preventDefault` stops working. I suggest the rule to only flag `preventDefault()` after the first `await`.
